test/cql-pytest/test_materialized_view.py

```diff
@@ -227,3 +226,31 @@ def test_is_not_operator_must_be_null(cql, test_keyspace):
                 cql.execute(f"CREATE MATERIALIZED VIEW {test_keyspace}.{mv} AS SELECT * FROM {table} WHERE p IS NOT 42 PRIMARY KEY (p)")
         finally:
             cql.execute(f"DROP MATERIALIZED VIEW IF EXISTS {test_keyspace}.{mv}")
+
+# Test that a view can be altered with synchronous_updates property and that
+# the synchronous updates code path is then reached for such view.
+def test_mv_synchronous_updates(cql, test_keyspace):
+    schema = 'p int, v text, primary key (p)'
+    with new_test_table(cql, test_keyspace, schema) as table:
+        def new_view():
+            return new_materialized_view(
+                cql, table, '*', 'v, p', 'v is not null and p is not null'
+            )
+
+        with new_view() as sync_mv, new_view() as async_mv:
+            # Make one view synchronous
+            cql.execute(f"ALTER MATERIALIZED VIEW {sync_mv} WITH synchronous_updates = true")
+
+            # Execute a query and inspect its tracing info
+            res = cql.execute(f"INSERT INTO {table} (p,v) VALUES (123, 'dog')", trace=True)
+            trace = res.get_query_trace()
```

Comment:
please check if this request doesn't generate a warning. I think you need to use res.one().get_query_trace()?

Reply:
indeed - I'll fix it in v2